metron-analytics/metron-statistics/src/main/java/org/apache/metron/statistics/StellarStatisticsFunctions.java

```diff
@@ -425,4 +428,74 @@ public Object apply(List<Object> args) {
       return result;
     }
   }
+
+  /**
+   * Calculates the statistical bin that a value falls in.
+   */
+  @Stellar(namespace = "STATS", name = "BIN"
+          , description = "Computes the bin that the value is in based on the statistical distribution."
+          , params = {
+          "stats - The Stellar statistics object"
+          , "value - The value to bin"
+          , "range? - A list of percentile bin ranges (excluding min and max) or a string representing a known and common set of bins.  " +
+          "For convenience, we have provided QUARTILE, QUINTILE, and DECILE which you can pass in as a string arg." +
+          " If this argument is omitted, then we assume a Quartile bin split."
```

Comment:
This is a list of bounds rather than ranges.  More precisely, the N'th bound specifies the closed upper bound of the N'th bin, where N is zero indexed; and if there is a list of M bounds then there are M+1 bins, numbered 0 through M. (Bin M is the bin for values greater than last bound in the list.)